source/nodejs/adaptivecards-designer/src/card-designer-surface.ts

```diff
@@ -830,8 +841,13 @@ export class CardDesignerSurface {
     setCardPayloadAsObject(payload: object) {
         this._serializationContext.clearEvents();
 
-        this.card.parse(payload, this._serializationContext);
-
+        if (!Array.isArray(payload)) {
+            this.card.parse(payload, this._serializationContext);
+            this._currentPayload = undefined;
+        } else {
+            // No need to parse the card here since templating will need to be applied first
+            this._currentPayload = payload;
```

Comment:
Nice to have: do you want to treat only one item in array the same with only one object?

Reply:
I don't think we want to do that. I think treating a one item array the same as one object would give users the impression that a one item array is a valid payload for the renderer.